tasks/hyrax-metadata-updates/index.js

```diff
@@ -0,0 +1,318 @@
+  } else {
+    shortName = metadata.Granule.Collection.ShortName;
+    version = metadata.Granule.Collection.VersionId;
+  }
+  // Query CMR for collection and retrieve entry title
+  const cmrInstance = new CMR({
+    provider: config.cmr.provider,
+    username: config.cmr.username,
+    password: config.cmr.passwordSecretName
+  });
+
+  const searchParams = {
+    short_name: shortName,
+    version: version
+  };
+
+  const result = await cmrInstance.searchCollections(searchParams);
+  // Verify that we have a valid result. If we don't then something is badly wrong
+  // and we should halt.
+  // Either the code is faulty or the provider is trying to ingest granules
+  // into a collection that doesn't exist
+  if (result.length === 0 || isUndefined(result[0].dataset_id)) {
+    throw new RecordDoesNotExist(`Unable to query parent collection entry title using short name ${shortName} and version ${version}`);
+  }
+  return result[0].dataset_id;
+}
+
+/**
+ * generatePath
+ *
+ * @param {Object} config - the config
+ * @param {Object} metadata - the dom
+ * @param {boolean} isUmmG - UMM-G or ECHO10
+ * @throws {Object} invalidArgumentException - if the env is not valid
+ * @returns {string} - the OPeNDAP path
+ */
+async function generatePath(config, metadata, isUmmG) {
+  const providerId = get(config.cmr, 'provider');
+  // Check if providerId is defined
+  if (isUndefined(providerId)) {
+    throw new InvalidArgument('Provider not supplied in configuration. Unable to construct path');
+  }
+  const entryTitle = await getEntryTitle(config, metadata, isUmmG);
+  return `providers/${providerId}/collections/${entryTitle}/granules/${getGranuleUr(metadata, isUmmG)}`;
+}
+
+/**
+ * generateHyraxUrl
+ *
+ * @param {Object} config - the config
+ * @param {Object} metadata - the dom
+ * @param {boolean} isUmmG - UMM-G or ECHO10
+ * @returns {string} - the hyrax url
+ */
+async function generateHyraxUrl(config, metadata, isUmmG) {
+  const path = await generatePath(config, metadata, isUmmG);
+  const url = new URL(`${generateAddress()}/${path}`);
+  return (url.href);
+}
+
+/**
+ * addHyraxUrlToUmmG
+ *
+ * @param {Object} metadata - the metadata dom
+ * @param {string} hyraxUrl - the hyrax url
+ * @returns {string} - the updated metadata containing a Hyrax URL
+ */
+function addHyraxUrlToUmmG(metadata, hyraxUrl) {
+  const metadataCopy = cloneDeep(metadata);
+
+  if (isUndefined(metadataCopy.RelatedUrls)) {
+    metadataCopy.RelatedUrls = [];
+  }
+  const url = {
+    URL: hyraxUrl,
+    Type: 'GET DATA',
+    Subtype: 'OPENDAP DATA',
+    Description: 'OPeNDAP request URL'
+  };
+  metadataCopy.RelatedUrls.push(url);
+
+  return JSON.stringify(metadataCopy, null, 2);
+}
+
+/**
+ * addHyraxUrlToEcho10
+ *
+ * @param {Object} metadata - the metadata dom
+ * @param {string} hyraxUrl - the hyrax url
+ * @returns {string} - the updated metadata containing a Hyrax URL
+ */
+async function addHyraxUrlToEcho10(metadata, hyraxUrl) {
```

Comment:
I don't think this function does anything that needs it to be `async`. Since `addHyraxUrl` doesn't `await` it either, it shouldn't be.